sqlmesh/core/context.py

```diff
@@ -2032,6 +2034,7 @@ def _create_engine_adapters(self, gateways: t.Optional[t.Set] = None) -> None:
 
     def _get_engine_adapter(self, gateway: t.Optional[str] = None) -> EngineAdapter:
         if gateway:
+            self._create_engine_adapters()
```

Comment:
Is this recomputed every time? Should we cache this?

Reply:
Yes, adapted this